graphql/resolver.go

```diff
@@ -35,6 +36,46 @@ func New(apiURL string) Config {
 			sc: dbConnector,
 		},
 	}
+	c.Directives.RequireHostAccess = func(ctx context.Context, obj interface{}, next graphql.Resolver, access HostAccessLevel) (interface{}, error) {
+		args, isStringMap := obj.(map[string]interface{})
+		if !isStringMap {
+			return nil, ResourceNotFound.Send(ctx, "host not specified")
+		}
+		hostId, hasHostId := args["hostId"].(string)
+		hostIdsInterface, hasHostIds := args["hostIds"].([]interface{})
```

Comment:
does it make sense to do this so you can skip the extra loop for `hostIds` 👀
```suggestion
		hostIds, hasHostIds := args["hostIds"].([]string)
```